main/utils/config.js

```diff
@@ -40,7 +66,42 @@ const getConfig = async () => {
   return config
 }
 
+const migrationMap = {
+  '1->2': (config) => {
+    config['_version'] = 2
+    if (config['advanced']['collect_usage_stats'] === undefined) {
+      config['advanced']['collect_usage_stats'] = true
+    }
+    return config
+  }
+}
+
+const migrate = (config, currentVersion, targetVersion) => {
+  const func = migrationMap[`${currentVersion}->${targetVersion}`]
+  if (func !=== undefined) {
+    return func(config)
+  }
+  log.error(`missing migrate function from ${currentVersion}->${targetVersion}`)
+}
+
+const maybeMigrate = async () => {
+  let config = await getConfig()
+  if (config['_version'] == LATEST_CONFIG_VERSION) {
+    return
+  }
+  if (config['_version'] > LATEST_CONFIG_VERSION) {
+    log.error('config file from the future')
+    return
+  }
+  for (const ver = config['_version']; ver <= LATEST_CONFIG_VERSION; ver++) {
```

Comment:
Looks like this will loop will run one extra time this way. It should be `ver < LATEST_CONFIG_VERSION`